File: documents/13-sdk/03-reference/python/main.py
from pydantic import BaseModel
from typing import Dict, Any

class ToolMetadata(BaseModel):
    id: str
    name: str
    category: str
    version: str
    permissions: list[str]

class Tool:
    metadata: ToolMetadata

    async def execute(self, input: Dict[str, Any]) -> Dict[str, Any]:
        raise NotImplementedError

class CatalogSearchTool(Tool):
    metadata = ToolMetadata(
        id="catalog.search",
        name="Search Catalog",
        category="catalog",
        version="1.0",
        permissions=["Public"]
    )

    async def execute(self, input: Dict[str, Any]) -> Dict[str, Any]:
        # In a real environment, this delegates to the Go Backend or MCP Bridge
        return {
            "summary": "Observation: Found 1 product",
            "data": {"items": [{"id": "1", "name": "Mock", "price": 10.0}]},
            "confidence": 0.99
        }

if __name__ == "__main__":
    print("SHOPWISE Python Reference AI Runtime")

File: documents/13-sdk/03-reference/go/main.go
package main

import (
	"context"
	"fmt"
)

// --- TOOL SDK ---

type ToolMetadata struct {
	ID          string
	Name        string
	Category    string
	Version     string
	Permissions []string
}

type Tool interface {
	Metadata() ToolMetadata
	InputSchema() any
	OutputSchema() any
	Execute(ctx context.Context, input any) (any, error)
}

// --- RETAIL SDK ---

type Product struct {
	ID    string
	Name  string
	Price float64
}

type CatalogProvider interface {
	Search(ctx context.Context, query string) ([]Product, error)
}

type RetailProvider interface {
	Catalog() CatalogProvider
}

// --- MOCK PROVIDER ---

type MockProvider struct{}

func (m *MockProvider) Catalog() CatalogProvider {
	return &MockCatalog{}
}

type MockCatalog struct{}

func (m *MockCatalog) Search(ctx context.Context, query string) ([]Product, error) {
	return []Product{
		{ID: "1", Name: "Mock Product", Price: 10.0},
	}, nil
}

func main() {
	fmt.Println("SHOPWISE Go Reference Backend Initialization...")
	
	provider := &MockProvider{}
	catalog := provider.Catalog()
	
	results, _ := catalog.Search(context.Background(), "test")
	fmt.Printf("Found %d products\n", len(results))
}
